src/Promitor.Integrations.AzureMonitor/AzureMonitorClient.cs

```diff
@@ -108,9 +109,13 @@ public AzureMonitorClient(AzureEnvironment azureCloud, string tenantId, string s
 
                 // Get the metric value according to the requested aggregation type
                 var requestedMetricAggregate = InterpretMetricValue(aggregationType, mostRecentMetricValue);
-
-                var measuredMetric = string.IsNullOrWhiteSpace(metricDimension) ? MeasuredMetric.CreateWithoutDimension(requestedMetricAggregate) : MeasuredMetric.CreateForDimension(requestedMetricAggregate, metricDimension, timeseries);
-                measuredMetrics.Add(measuredMetric);
+                try {
+                    var measuredMetric = string.IsNullOrWhiteSpace(metricDimension) ? MeasuredMetric.CreateWithoutDimension(requestedMetricAggregate) : MeasuredMetric.CreateForDimension(requestedMetricAggregate, metricDimension, timeseries);
+                    measuredMetrics.Add(measuredMetric);
+                } catch (ArgumentException) {
+                    _logger.LogWarning("{MetricName} has return a time series with empty value for {Dimension} and the measurements will be dropped", metricName, metricDimension); 
+                    _logger.LogDebug("The violating time series has content {}", JsonConvert.SerializeObject(timeseries)); 
```

Comment:
Do we really need this? What is the goal here? To analyse? If so, doesn't the Azure Monitor logs help?

(also, the format is wrong as {} has no value)

Reply:
I think it'd be useful for our case, to analyze what Azure's returning exactly. It would then inform us on what actions we can take(if any). I believe logging here is more useful than looking in Azure Monitor logs, because
-  we would then know exactly to search for. We would look log with such message, and +/- a few seconds to gather all the other relevant info we need. 
-  we like to search for logs in a centralized place instead of looking across multiple places at the same time.

Reply:
It's OK, it's debug anyway and was just curious why